transports/webrtc/src/udp_mux.rs

```diff
@@ -140,9 +141,24 @@ impl UDPMuxNewAddr {
 
     /// Returns a muxed connection if the `ufrag` from the given STUN message matches an existing
     /// connection.
-    fn conn_from_stun_message(&self, buffer: &[u8], addr: &SocketAddr) -> Option<UDPMuxConn> {
+    fn conn_from_stun_message(
+        &self,
+        buffer: &[u8],
+        addr: &SocketAddr,
+    ) -> Option<Result<UDPMuxConn, ConnQueryError>> {
         match ufrag_from_stun_message(buffer, true) {
-            Ok(ufrag) => self.conns.get(&ufrag).map(Clone::clone),
+            Ok(ufrag) => {
+                if let Some(conn) = self.conns.get(&ufrag) {
+                    let associated_addrs = conn.get_addresses();
+                    // This basically ensures only one address is registered per ufrag.
+                    if associated_addrs.is_empty() || associated_addrs.contains(addr) {
+                        return Some(Ok(conn.clone()));
+                    } else {
+                        return Some(Err(ConnQueryError::UfragAlreadyTaken { associated_addrs }));
+                    }
+                }
+                None
```

Comment:
Unless I misunderstand https://github.com/libp2p/specs/commit/7ce6213ee0998660032b853d7c64f3a3dae312b7, I don't think this actually reflects the spec.

Can we not as @mxinden suggested change the key of the hashmap and thus allow the same ufrag for different connections?

Reply:
> Can we not as @mxinden suggested change the key of the hashmap and thus allow the same ufrag for different connections?

No, unfortunately. That would require us to fork and modify webrtc-rs library.

Reply:
@melekes can you expand on this? I am assuming that we can not index by ufrag and origin address as we don't have the latter available at all times?

The original suggestion:

``` rust
conn: HashMap<(String /* ufrag */, SocketAddr /* remote addr */), UDPMuxConn>,
```

Reply:
Ah, got it. Thanks for explaining @melekes. That makes sense. If I am not missing something we can leave it as is with your patch https://github.com/libp2p/rust-libp2p/pull/2622/commits/226f2930daf9a2614c216546c6084de646a9f26d. //CC @thomaseizinger

Reply:
Sounds good to me.